### PR TITLE
[Issue 11339] Pulsar Admin List Subscription lists only subscriptions created for Partition-0 when partition specific subscriptions are created

### LINKED ISSUE #11339: Pulsar Admin List Subscription lists only subscriptions created for Partition-0 when partition specific subscriptions are created
**Describe the bug**

Pulsar Admin List Subscription lists only subscriptions created for Partition-0 when partition specific subscriptions are created

**To Reproduce**
Pulsar Version: 2.7.0 and 2.7.2.

	1. Create a topic with N partitions [./pulsar-admin topics create-partitioned-topic public/default/nodatahere1 -p 3]
	2. Create Partition specific Subscriptions [ Example:  pulsar-admin topics create-subscription -s 0_abc -m earliest persistent://public/default/nodatahere1-partition-0, pulsar-admin topics create-subscription -s 0_abc_1 -m earliest persistent://public/default/nodatahere1-partition-0, pulsar-admin topics create-subscription -s 1_abc -m earliest persistent://public/default/nodatahere1-partition-1]
	3. Listing partitions specific subscription works fine [pulsar-admin topics subscriptions persistent://public/default/nodatahere1-partition-0, pulsar-admin topics subscriptions persistent://public/default/nodatahere1-partition-1]. For partition-0, 0_abc and 0_abc_1 is listed. For partition-1, 1_abc is listed. 
	4. List the partitions at the topic level [pulsar-admin topics subscriptions persistent://public/default/nodatahere1] 
		a. List partitions lists "0_abc" and "0_abc_1" alone. It does not list "1_abc". 

Expectation: It should list all partition specific topics or None of the partition specific topics.  Currently, if we have a subscription "abc_all" that is across the partitions, then list subscriptions will list "0_abc", "0_abc_1", "abc_all". 

**Expected behavior**
It should list all partition specific topics or None of the partition specific topics.  Currently, if we have a subscription "abc_all" that is across the partitions, then list subscriptions will list "0_abc", "0_abc_1", "abc_all". 

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by Technoboy-**
Fix #11339.

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -1030,34 +1030,35 @@ protected void internalGetSubscriptions(AsyncResponse asyncResponse, boolean aut
...
+                        for (int i = 0; i < partitionMetadata.partitions; i++) {
+                            CompletableFuture<List<String>> subscriptionsAsync = pulsar().getAdminClient().topics()
+                                    .getSubscriptionsAsync(topicName.getPartition(i).toString());
+                            subscriptionFutures.add(subscriptionsAsync.thenApply(r -> subscriptions.addAll(r)));
+                        }
+                        FutureUtil.waitForAll(subscriptionFutures).whenComplete((r, ex) -> {
+                            if (ex != null) {
+                                log.warn("[{}] Failed to get list of subscriptions for {}: {}", clientAppId(),
+                                        topicName, ex.getMessage());
+                                if (ex instanceof PulsarAdminException) {
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1044]: In this case maybe throw NotFoundException: Topic partitions were not yet created
Example：
```
./bin/pulsar-admin topics create-partitioned-topic public/default/nodatahere -p 3
./bin/pulsar-admin topics create-subscription -s sub_0 -m earliest persistent://public/default/nodatahere-partition-0
./bin/pulsar-admin topics create-subscription -s sub_1 -m earliest persistent://public/default/nodatahere-partition-0
./bin/pulsar-admin topics create-subscription -s sub_2 -m earliest persistent://public/default/nodatahere-partition-1
```
Waiting for brokerDeleteInactiveTopicsFrequencySeconds (default: 60s) or call PersistentTopic.checkGC() if the deleteMode == InactiveTopicDeleteMode.delete_when_no_subscriptions,  topic nodatahere-partition-2 will be delete.

```
./bin/pulsar-admin topics subscriptions persistent://public/default/nodatahere-partition-2
./bin/pulsar-admin topics subscriptions persistent://public/default/nodatahere
```

**Review by Demogorgon314 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1044]: Thanks for kindly reviewing.

**Comment by Technoboy-** [comment]
Hi @Demogorgon314:
I reopen a [new pr](https://github.com/apache/pulsar/pull/11355), please help to review, thanks very much .

**Pull request closed without merging**